### PR TITLE
infra: Upgrade setuptools to version explicitly required by pyinstaller

### PR DESCRIPTION
**Pull request by DaveLak**
Fixes Issue: #11742

## TL;DR
Pyinstaller requires `'setuptools>=42.0.0'` to properly detect "hidden import" module dependencies of a particular package that it is unable to determine through conventional means.

`'setuptools==42.0.2'` was chosen here to:
- keep with the convention used in this file of pinning specific versions as apposed to minimums/ranges.
- introduce the smallest major version bump possible to minimize introduction of changes that are not specific to the pyinstaller issue this aims to fix.

I've tested this on a few projects locally but wasn't sure how to effectively test against all of the python projects locally, so I haven't :(

### Broken Builds This Fixes
These are a few projects (note that there might be more, this is a subset I tested locally)

- https://introspector.oss-fuzz.com/project-profile?project=connexion
- https://introspector.oss-fuzz.com/project-profile?project=adal
- https://introspector.oss-fuzz.com/project-profile?project=gcp-python-cloud-storage
- https://introspector.oss-fuzz.com/project-profile?project=g-cloud-logging-py
- https://introspector.oss-fuzz.com/project-profile?project=h5py
- https://introspector.oss-fuzz.com/project-profile?project=httplib2
- https://introspector.oss-fuzz.com/project-profile?project=msal
- https://introspector.oss-fuzz.com/project-profile?project=looker-sdk
- https://introspector.oss-fuzz.com/project-profile?project=pyzmq

### CODE REVIEW

**Comment by DonggeLiu** [comment]
@oliverchang @jonathanmetzman Is now a good time to take this PR?
Thanks!

**Comment by DaveLak** [comment]
I ran the build check locally after updating this branch and it looks good what I tested:

- `bs4`: Passed
- `lxml`: Will fail until #11709 or similar is merged
- `xlsxwriter`: Passed
- `pandas`: Passed; it took a while to build and logged `WARNING: Cannot find <somelib>.so` many times but that appears to be normal for the project[^1]

[^1]: https://oss-fuzz-build-logs.storage.googleapis.com/log-b5b4a4fa-9c5b-4a74-9cb4-ab2f00fa8462.txt

**Comment by DaveLak** [comment]
It looks like the trial build timed out with 4 projects not run, but no failed builds. Here's what I got running `check_build` on the 4 projects locally:

- [x] `decorator`: Passed
- [x] `gprof2dot`: Passed
- [ ] `lxml`: Failed as expected; [see my previous comment](https://github.com/google/oss-fuzz/pull/11743#issuecomment-2070516880)
- [x] `pyasn1`: Passed

Logs link: https://storage.cloud.google.com/oss-fuzz-trialbuild-logs/log-6ebcda2e-6f62-4ff5-a51e-ed3d62e55453.txt

<details><summary>Tail of the Log -- "Build result" section</summary>

```shell
Step #1: INFO:root:----------------------------Build result----------------------------
Step #1: INFO:root:Trial build end time: 2024-05-02 01:15:08.096928
Step #1: INFO:root:Failed project, Statuses, Logs
Step #1: INFO:oauth2client.transport:Refreshing due to a 401 (attempt 1/2)
Step #1: INFO:root:[2024-05-01 20:19:28.955805] Remaining builds: 6, defaultdict(<class 'list'>, {'decorator': ['6f835dc2-a9dc-435a-aefe-4f660a6e44de'], 'gprof2dot': ['f7b3d643-28a6-41f4-b930-a4046d53ed84'], 'lxml': ['cda13dad-04b2-428b-a69c-6abc75612976'], 'pandas': ['f82feb7a-0262-444e-aa4a-011b15cfee52', '528811fb-9c53-4d9c-a89e-38786d02e6c1'], 'pyasn1': ['b81362b6-53e0-4a4e-8605-01d8dbc5982c'], 'pyxdg': ['022eba78-f555-4b92-9c1c-d490dfed0865']})
Step #1: INFO:oauth2client.transport:Refreshing due to a 401 (attempt 1/2)
Step #1: INFO:root:[2024-05-01 21:19:25.776877] Remaining builds: 4, defaultdict(<class 'list'>, {'decorator': ['6f835dc2-a9dc-435a-aefe-4f660a6e44de'], 'gprof2dot': ['f7b3d643-28a6-41f4-b930-a4046d53ed84'], 'lxml': ['cda13dad-04b2-428b-a69c-6abc75612976'], 'pyasn1': ['b81362b6-53e0-4a4e-8605-01d8dbc5982c']})
Step #1: INFO:oauth2client.transport:Refreshing due to a 401 (attempt 1/2)
Step #1: INFO:root:[2024-05-01 22:19:22.854935] Remaining builds: 4, defaultdict(<class 'list'>, {'decorator': ['6f835dc2-a9dc-435a-aefe-4f660a6e44de'], 'gprof2dot': ['f7b3d643-28a6-41f4-b930-a4046d53ed84'], 'lxml': ['cda13dad-04b2-428b-a69c-6abc75612976'], 'pyasn1': ['b81362b6-53e0-4a4e-8605-01d8dbc5982c']})
Step #1: INFO:oauth2client.transport:Refreshing due to a 401 (attempt 1/2)
Step #1: INFO:root:[2024-05-01 23:19:26.141743] Remaining builds: 4, defaultdict(<class 'list'>, {'decorator': ['6f835dc2-a9dc-435a-aefe-4f660a6e44de'], 'gprof2dot': ['f7b3d643-28a6-41f4-b930-a4046d53ed84'], 'lxml': ['cda13dad-04b2-428b-a69c-6abc75612976'], 'pyasn1': ['b81362b6-53e0-4a4e-8605-01d8dbc5982c']})
Step #1: INFO:oauth2client.transport:Refreshing due to a 401 (attempt 1/2)
Step #1: INFO:root:[2024-05-02 00:19:25.894097] Remaining builds: 4, defaultdict(<class 'list'>, {'decorator': ['6f835dc2-a9dc-435a-aefe-4f660a6e44de'], 'gprof2dot': ['f7b3d643-28a6-41f4-b930-a4046d53ed84'], 'lxml': ['cda13dad-04b2-428b-a69c-6abc75612976'], 'pyasn1': ['b81362b6-53e0-4a4e-8605-01d8dbc5982c']})
Step #1: INFO:oauth2client.transport:Refreshing due to a 401 (attempt 1/2)
Step #1: INFO:root:[2024-05-02 01:00:11.937814] Warning: trial build may time out in 15 minutes.
Step #1: Remaining builds: 4/235, defaultdict(<class 'list'>, {'decorator': ['6f835dc2-a9dc-435a-aefe-4f660a6e44de'], 'gprof2dot': ['f7b3d643-28a6-41f4-b930-a4046d53ed84'], 'lxml': ['cda13dad-04b2-428b-a69c-6abc75612976'], 'pyasn1': ['b81362b6-53e0-4a4e-8605-01d8dbc5982c']}).
Step #1: Failed builds: 0/235, {}
```

</details>
